typescript/cli/src/deploy/utils.ts

```diff
@@ -86,12 +86,30 @@ export async function runDeployPlanStep({
   );
 
   if (skipConfirmation) return;
+  await confirmExistingMailbox(context, chain);
   const isConfirmed = await confirm({
     message: 'Is this deployment plan correct?',
   });
   if (!isConfirmed) throw new Error('Deployment cancelled');
 }
 
+async function confirmExistingMailbox(
+  context: CommandContext,
+  chain: ChainName,
+) {
+  const addresses = await context.registry.getChainAddresses(chain);
+  if (addresses?.mailbox) {
+    const isConfirmed = await confirm({
+      message: `Mailbox has already exists at ${addresses.mailbox}, registry artifacts will be overwritten`,
```

Comment:
something like this to better inform what's happening - probs needs some workshopping on the msg
```suggestion
      message: `Mailbox already exists at ${addresses.mailbox}. Are you sure you want to deploy a new mailbox and overwrite existing registry artifacts?`,
```